flowmachine/flowmachine/core/cache.py

```diff
@@ -238,14 +247,17 @@ def reset_cache(connection: "Connection", redis: StrictRedis) -> None:
     # they point to a cache table and hence are a duplicate of a Query entry which
     # will also point to that table.
 
-    qry = f"SELECT tablename FROM cache.cached WHERE NOT class='Table'"
-    tables = connection.fetch(qry)
     with connection.engine.begin() as trans:
+        qry = f"SELECT tablename FROM cache.cached WHERE NOT class='Table'"
+        tables = connection.fetch(qry)
         trans.execute("SELECT setval('cache.cache_touches', 1)")
         for table in tables:
             trans.execute(f"DROP TABLE IF EXISTS cache.{table[0]} CASCADE")
-        trans.execute("TRUNCATE cache.cached CASCADE")
         trans.execute("TRUNCATE cache.dependencies CASCADE")
+        if protect_table_objects:
+            trans.execute(f"DELETE FROM cache.cached WHERE NOT class='Table'")
```

Comment:
If we have a `Table` object which points to a cache table (e.g. cache a daily location query in cache table `cache.xf9ff4884de630fbc5ba351ed2693b107`, and then create `daily_location_cache_table = Table("cache.xf9ff4884de630fbc5ba351ed2693b107")`), then the cache table will be dropped but the row in `cache.cached` corresponding to `daily_location_cache_table` won't be deleted because it has class=`Table`. This would lead to the opposite situation from before, where `daily_location_cache_table.is_stored == False` but `Query.get_stored()` includes `daily_location_cache_table`.

Would `"DELETE FROM cache.cached WHERE schema='cache'"` work? In fact, could we select the tables to drop using `qry = f"SELECT tablename FROM cache.cached WHERE schema='cache'"`, to remove the need to handle Table objects differently?

Reply:
Excellent suggestion!